filament/backend/src/metal/MetalContext.mm

```diff
@@ -24,9 +24,30 @@
 namespace backend {
 namespace metal {
 
+void presentDrawable(void* user) {
+    id<CAMetalDrawable> drawable = (id<CAMetalDrawable>) user;
+    [drawable present];
+    [drawable release];
+}
+
 id<CAMetalDrawable> acquireDrawable(MetalContext* context) {
     if (!context->currentDrawable) {
-        context->currentDrawable = [context->currentSurface->layer nextDrawable];
+        // The drawable is retained here and will be released either:
```

Comment:
Can the other backends do something meaningful with the new callback? If not, we should probably clearly document which backends support this feature.

Reply:
That's a good point. I should certainly document that only Metal supports this at the moment. Other backends could implement the functionality, but I don't think there's a use case for that.